infra/relayer/src/relayer.js

```diff
@@ -334,6 +337,17 @@ class Relayer {
       return res.status(400).send({ errors: ['Invalid function signature'] })
     }
 
+    // 2.1 Deep check that the potentialy nested transaction are calling
+    // approved methods. For now, we don't fail on this, since we might
+    // have the whitelist too tight. We'll check the logs after a while
+    // and see if there is anything we should add.
+    try {
+      this.validator.validate(PROXY_HARDCODE, txData)
+    } catch (e) {
+      logger.error('Error in transaction validator', e)
+      throw e
```

Comment:
maybe do a `res.status(500).send [...]` instead of killing the relayer

Reply:
Thank goodness for code reviews!

I've pulled out the throw - that shouldn't have been there. It will now correctly log verifier crashes, but not crash the relayer.